Sources/PortalSwift/Core/Mpc/Mobile/EnclaveMobileWrapper.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  EnclaveMobileWrapper.swift
+//  PortalSwift
+//
+//  Created by Rami Shahatit on 2/19/25.
+//
+
+import Foundation
+
+class EnclaveMobileWrapper: MPCMobile {
+  private let requests: PortalRequestsProtocol
+
+  init(
+    requests: PortalRequestsProtocol = PortalRequests()
+  ) {
+    self.requests = requests
+  }
+
+  // Override sign method to use HTTP endpoint
+  func MobileSign(
+    _ apiKey: String?,
+    _: String?,
+    _ signingShare: String?,
+    _ method: String?,
+    _ params: String?,
+    _ rpcURL: String?,
+    _ chainId: String?,
+    _ metadata: String?
+  ) async -> String {
+    guard let apiKey = apiKey,
+          let signingShare = signingShare,
+          let method = method,
+          let params = params,
+          let rpcURL = rpcURL,
+          let chainId = chainId,
+          let metadata = metadata
+    else {
+      return encodeErrorResult(id: "INVALID_PARAMETERS", message: "Invalid parameters provided")
+    }
+
+    guard let url = URL(string: "https://mpc-client.portalhq.io/v1/sign") else {
+      return encodeErrorResult(id: "INVALID_URL", message: "Invalid URL")
```

Comment:
this likely shouldnt be an mpc error rather it should just be a regular swift error